app/include/zmk/events/layer-state-changed.h

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#pragma once
+
+#include <zephyr.h>
+#include <zmk/event-manager.h>
+
+struct layer_state_changed {
+    struct zmk_event_header header;
+    u32_t layer;
+    bool state;
+    s64_t timestamp;
+};
+
+ZMK_EVENT_DECLARE(layer_state_changed);
```

Comment:
Check out `app/include/zmk/events/modifiers-state-changed.h`, for an example of a more complete "event factory function", that I think we should be using for new event types in the future, instead of requiring folks wishing to fire the event to set the fields themselves.

Better encapsulation, and easier API, so you can then do:

```
ZMK_EVENT_RAISE(create_layer_state_changed(layer, state));
```

And the factory would set the `timestamp` to current time, etc.